src/web/client/extension.ts

```diff
@@ -314,6 +292,49 @@ export function processWorkspaceStateChanges(context: vscode.ExtensionContext) {
     );
 }
 
+export function processActiveTextEditorChange(context: vscode.ExtensionContext) {
+    context.subscriptions.push(
+        vscode.window.onDidChangeActiveTextEditor((editor) => {
+            if (editor) {
+                const document = editor.document;
+                const entityInfo: IEntityInfo = {
+                    entityId: getFileEntityId(document.uri.fsPath),
+                    entityName: getFileEntityName(document.uri.fsPath),
+                    rootWebPageId: getFileRootWebPageId(document.uri.fsPath),
+                };
+                if (entityInfo.entityId && entityInfo.entityName && isCoPresenceEnabled()) {
+                    // sending message to webworker event listener for Co-Presence feature
+                    sendingMessageToWebWorkerForCoPresence(entityInfo);
+                }
+
+                WebExtensionContext.quickPickProvider.refresh();
+            }
+        })
+    );
+}
+
+function sendingMessageToWebWorkerForCoPresence(entityInfo: IEntityInfo) {
```

Comment:
Can we move this function to copresence utility file - this does not belong in entry file for extension

Reply:
done!